pkg/typesystem/error.go

```diff
@@ -9,66 +8,100 @@ import (
 )
 
 var (
-	ErrModelNotFound          = errors.New("authorization model not found")
-	ErrDuplicateTypes         = errors.New("an authorization model cannot contain duplicate types")
-	ErrInvalidSchemaVersion   = errors.New("invalid schema version")
-	ErrInvalidModel           = errors.New("invalid authorization model encountered")
-	ErrRelationUndefined      = errors.New("undefined relation")
-	ErrObjectTypeUndefined    = errors.New("undefined object type")
-	ErrInvalidUsersetRewrite  = errors.New("invalid userset rewrite definition")
-	ErrReservedKeywords       = errors.New("self and this are reserved keywords")
-	ErrCycle                  = errors.New("an authorization model cannot contain a cycle")
-	ErrNoEntrypoints          = errors.New("no entrypoints defined")
-	ErrNoEntryPointsLoop      = errors.New("potential loop")
+	// ErrModelNotFound is returned when an authorization model is not found.
+	ErrModelNotFound = errors.New("authorization model not found")
+
+	// ErrDuplicateTypes is returned when an authorization model contains duplicate types.
+	ErrDuplicateTypes = errors.New("an authorization model cannot contain duplicate types")
+
+	// ErrInvalidSchemaVersion is returned for an invalid schema version in the authorization model.
+	ErrInvalidSchemaVersion = errors.New("invalid schema version")
+
+	// ErrInvalidModel is returned when encountering an invalid authorization model.
+	ErrInvalidModel = errors.New("invalid authorization model encountered")
+
+	// ErrRelationUndefined is returned when encountering an undefined relation in the authorization model.
+	ErrRelationUndefined = errors.New("undefined relation")
+
+	// ErrObjectTypeUndefined is returned when encountering an undefined object type in the authorization model.
+	ErrObjectTypeUndefined = errors.New("undefined object type")
+
+	// ErrInvalidUsersetRewrite is returned for an invalid userset rewrite definition.
+	ErrInvalidUsersetRewrite = errors.New("invalid userset rewrite definition")
+
+	// ErrReservedKeywords is returned when using reserved keywords "self" and "this".
+	ErrReservedKeywords = errors.New("self and this are reserved keywords")
+
+	// ErrCycle is returned when an authorization model contains a cycle.
+	ErrCycle = errors.New("an authorization model cannot contain a cycle")
+
+	// ErrNoEntrypoints is returned when no entrypoints are defined in the authorization model.
```

Comment:
```suggestion
	// ErrNoEntrypoints is returned when an authorization model contains a specific objectType and relation that is not reachable from a direct edge (i.e. from another objectType).
```

Example:

<img width="1580" alt="image" src="https://github.com/openfga/openfga/assets/5374887/cbed5293-11c7-4e7d-bf7c-5cb90869b9fe">